wincent/liferay-portal
# LPS-113561 Avoid unnecessary use of `Liferay.provide` in knowledge-base-web

PR 6 of the 6 described here: https://github.com/wincent/liferay-portal/pull/320#issuecomment-641223200

cc @julien

Original message follows.

---

As part of [LPS-113561](https://issues.liferay.com/browse/LPS-113561) we eventually want to deprecate the `Liferay.provide` function. In this change we avoid declaring the `<portlet:namespace />deleteFileEntry` function as a global on the `window` object.

What's interesting is that before doing any change if you test this functionality, you'll see that it currently throws an error as can be seen in the following [screenshot](https://user-images.githubusercontent.com/5572/84125897-7fad1d00-aa3d-11ea-8232-fd5b372afe54.png)
this has been fixed in this changed.

To test this change:

- Navigate to "Content and Data > Knowledge Base"
- Click on the "+" button and select "Basic Article"
- Enter a name for your article
- Add text in the content field
- Scroll down to the "Attachments" section and click on the "Select Files" button
- Upload a file (for example an image or a text file)
- Scroll down and click the "Publish" button
- From the list of knowledge base articles, click on the kebab menu to the right of article you created previously, and select "Edit"
- Scroll down to the "Attachments" section
- Click on the "Delete" link to the right of the file you uploaded previously
- After confirming the file you attached should disappear
- Scroll down and click on the "Publish" button

As a result the file you uploaded previously should no longer be visible, and no JS errors should appear in your browser's console.
